File: java/src/main/java/ezocalc/structure/Value.java
package ezocalc.structure;

import java.math.BigDecimal;

public class Value implements Bloc{

	BigDecimal value;

	public Value(String value) throws SyntaxError {
		try {
			this.value = BigDecimal.valueOf(Double.valueOf(value));
		} catch (Exception e) {
			throw new SyntaxError("Invalid value : " + value);
		}
	}

	public Value(BigDecimal value) throws SyntaxError {
		this.value = value;
	}

	public BigDecimal getValue() {
		return value;
	}

	public boolean is_operand() {
		return false;
	}

}

File: lambda/lambdas/ezocalc/value.py
#!/usr/bin/env python3

from numbers import Number
from decimal import *
import ezocalc.bloc


class Value(ezocalc.bloc.Bloc):

    _value: Decimal

    def __init__(self, value):
        try:
            self._value = Decimal(value)
        except InvalidOperation:
            raise SyntaxError(f'Invalid value : {value}')

    def get_value(self) -> Number:
        return self._value